python/tank/bootstrap/manager.py

```diff
@@ -221,11 +244,50 @@ def set_progress_callback(self, callback):
         """
         self._progress_cb = callback
 
-    def bootstrap_engine(self, engine_name, entity=None):
+    def bootstrap_engine(self,
```

Comment:
Manne should elaborate on this, but, as far as I know, usage of a property will make it easy to add control of the synchronous or asynchronous bootstrap logic using a config file value.

Also note that keeping both sections of the bootstrap logic code in the same method makes it easy to fallback from asynchronous to synchronous when Qt is not available.